### ISSUE TITLE
`join` throws `AttributeError` when `slack` parameter is nonzero and there are no overlaps within radius.

### ISSUE THREAD

**Issue by edridgedsouza** [biocore-ntnu/pyranges#198]
It appears that `pr.join()` will throw an `AttributeError` if the `slack` parameter is nonzero and there are no overlaps found within the slack radius.

Compare the outputs of these reprexes:

```
r1 = pr.from_dict({'Chromosome': ['1','2'],
                  'Start': [10, 20],
                  'End': [14, 24],
                  'id': ['a','b']})

r2 = pr.from_dict({'Chromosome': ['3', '4'],
                  'Start': [0, 26],
                  'End': [9, 30],
                  'id': ['c', 'd']})

r1.join(r2, slack=0, report_overlap=True)

### Returns blank df
```

```
r1 = pr.from_dict({'Chromosome': ['1','2'],
                  'Start': [10, 20],
                  'End': [14, 24],
                  'id': ['a','b']})

r2 = pr.from_dict({'Chromosome': ['3', '4'],
                  'Start': [0, 26],
                  'End': [9, 30],
                  'id': ['c', 'd']})

r1.join(r2, slack=20, report_overlap=True)

## Throws AttributeError
```

```
r1 = pr.from_dict({'Chromosome': ['1','2'],
                  'Start': [10, 20],
                  'End': [14, 24],
                  'id': ['a','b']})

r2 = pr.from_dict({'Chromosome': ['2', '4'],
                  'Start': [0, 26],
                  'End': [9, 30],
                  'id': ['c', 'd']})

r1.join(r2, slack=0, report_overlap=True)

# Returns blank df
```

```
r1 = pr.from_dict({'Chromosome': ['1','2'],
                  'Start': [10, 20],
                  'End': [14, 24],
                  'id': ['a','b']})

r2 = pr.from_dict({'Chromosome': ['2', '4'],
                  'Start': [0, 26],
                  'End': [9, 30],
                  'id': ['c', 'd']})

r1.join(r2, slack=10, report_overlap=True)

# Throws AttributeError
```

```
r1 = pr.from_dict({'Chromosome': ['1','2'],
                  'Start': [10, 20],
                  'End': [14, 24],
                  'id': ['a','b']})

r2 = pr.from_dict({'Chromosome': ['2', '4'],
                  'Start': [0, 26],
                  'End': [9, 30],
                  'id': ['c', 'd']})

r1.join(r2, slack=20, report_overlap=True)

## Returns the following df:

# +--------------+-----------+-----------+------------+-----------+-------+
# |   Chromosome |     Start |       End | id         |   Start_b | +3    |
# |   (category) |   (int32) |   (int32) | (object)   |   (int32) | ...   |
# |--------------+-----------+-----------+------------+-----------+-------|
# |            2 |        20 |        24 | b          |         0 | ...   |
# +--------------+-----------+-----------+------------+-----------+-------+
# Unstranded PyRanges object has 1 rows and 8 columns from 1 chromosomes.
# For printing, the PyRanges was sorted on Chromosome.
# 3 hidden columns: End_b, id_b, Overlap
```

**Comment by endrebak** [comment]
Thanks for reporting. Will fix after work

**Comment by endrebak** [comment]
Should be fixed in `0.0.101` which is out now :)
